examples/flutter_gallery/test/demo/material/menu_demo_test.dart

```diff
@@ -0,0 +1,148 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:typed_data';
+import 'dart:ui' as ui;
+
+import 'package:flutter/material.dart';
+import 'package:flutter_gallery/demo/material/menu_demo.dart';
+import 'package:flutter_gallery/gallery/themes.dart';
+import 'package:flutter_test/flutter_test.dart';
+
+import 'package:flutter/rendering.dart';
+
+void main() {
+  testWidgets('Menu icon satisfies accessibility contrast ratio guidelines, light mode', (WidgetTester tester) async {
+    await tester.pumpWidget(MaterialApp(
+      theme: kLightGalleryTheme,
+      home: const MenuDemo(),
+    ));
+
+    await expectLater(tester, meetsGuideline(textContrastGuideline));
+
+    final List<Element> icons = find.byWidgetPredicate((Widget widget) => widget is Icon).evaluate().toList();
+
+    await expectLater(tester, meetsGuideline(CustomContrastGuideline(elements: icons)));
+  });
+
+  testWidgets('Menu icon satisfies accessibility contrast ratio guidelines, dark mode', (WidgetTester tester) async {
```

Comment:
Rather than copying this into a test, you could create this as a new option in the flutter_test package, or allow configuring the existing accessibility guideline

Reply:
Good idea. I'll try to extract it into a test.